src/main/java/com/fdmgroup/forex/DataLoader.java

```diff
@@ -33,4 +63,30 @@ private void createRoles() {
 			roleRepo.save(new Role(defaultRole));
 	}
 
+	@Transactional
+	private void createSampleData() {
+		Role role = roleRepo.findByRole(defaultRole).get();
+
+		Currency hkd = currencyRepo.findById(usdCode).orElseGet(() -> {
```

Comment:
This should be `hkd = currencyRepo.findById(hkdCode)` -> currently reads `usdCode`